tests/config/test_config.py

```diff
@@ -0,0 +1,95 @@
+"""Test config."""
+
+import os
+import tempfile
+
+import pytest
+from pydantic import ValidationError
+
+from ocf_data_sampler.config.load import load_yaml_configuration
+from ocf_data_sampler.config.model import Configuration
+from ocf_data_sampler.config.save import save_yaml_configuration
+
+
+def test_default():
+    """
+    Test default pydantic class
+    """
+
+    _ = Configuration()
+
+
+def test_yaml_load_test_config(test_config_filename):
+    """Test that yaml loading works for 'test_config.yaml'"""
+
+    config = load_yaml_configuration(test_config_filename)
+
+    assert isinstance(config, Configuration)
+
+
+def test_yaml_save():
+    """
+    Check a configuration can be saved to a yaml file
+    """
+
+    with tempfile.NamedTemporaryFile(suffix=".yaml") as fp:
+        filename = fp.name
+
+        # check that temp file can't be loaded
+        with pytest.raises(TypeError):
+            _ = load_yaml_configuration(filename)
+
+        # save default config to file
+        save_yaml_configuration(Configuration(), filename)
```

Comment:
This is too simple to be a useful test. Could we load the config in `test_yaml_load_test_config()` and test saving that?

Reply:
Good to check the saving works right?

Reply:
Yeh definitely, but the default config has no input sources and so we are essentially saving out `{gsp: None, satellite: None, nwp:none}`. If we instead save out the config which we load in the `test_yaml_load_test_config()` there are some inputs to be saved